it/controllers/benefits/medical/EducationalServicesAmountControllerISpec.scala

```diff
@@ -0,0 +1,462 @@
+          }
+
+          implicit def document: () => Document = () => Jsoup.parse(result.body)
+
+          import Selectors._
+          import user.commonExpectedResults._
+
+          "has an OK status" in {
+            result.status shouldBe OK
+          }
+
+          titleCheck(user.specificExpectedResults.get.expectedTitle)
+          h1Check(user.specificExpectedResults.get.expectedHeading)
+          captionCheck(expectedCaption)
+          textOnPageCheck(previousExpectedContent, contentSelector)
+          textOnPageCheck(amountHint, hintTextSelector)
+          textOnPageCheck(poundPrefixText, poundPrefixSelector)
+          inputFieldCheck(amountInputName, inputSelector)
+          inputFieldValueCheck("300", inputSelector)
+          buttonCheck(continue, continueButtonSelector)
+          formPostLinkCheck(continueLink, continueButtonFormSelector)
+          welshToggleCheck(user.isWelsh)
+        }
+      }
+    }
+
+    "redirect to another page when the request is valid but they aren't allowed to view the page and" should {
+      "Redirect user to the check your benefits page with no cya data in session" in {
+        lazy val result: WSResponse = {
+          dropEmploymentDB()
+          userDataStub(userData(fullEmploymentsModel(hmrcEmployment = Seq(employmentDetailsAndBenefits(fullBenefits)))), nino, taxYearEOY)
+          authoriseAgentOrIndividual(isAgent = false)
+          urlGet(pageUrl(taxYearEOY), follow = false, headers = Seq(HeaderNames.COOKIE -> playSessionCookies(taxYearEOY)))
+        }
+
+        result.status shouldBe SEE_OTHER
+        result.header("location") shouldBe Some(CheckYourBenefitsController.show(taxYearEOY, employmentId).url)
+      }
+
+      "Redirect user to the tax overview page when in year" in {
+        lazy val result: WSResponse = {
+          dropEmploymentDB()
+          userDataStub(userData(fullEmploymentsModel(hmrcEmployment = Seq(employmentDetailsAndBenefits(fullBenefits)))), nino, taxYearEOY)
+          insertCyaData(employmentUserData(isPrior = false, cyaModel(hmrc = true, benefitsWithNoBenefitsReceived)), userRequest)
+          authoriseAgentOrIndividual(isAgent = false)
+          urlGet(pageUrl(taxYear), follow = false, headers = Seq(HeaderNames.COOKIE -> playSessionCookies(taxYearEOY)))
+        }
+
+        result.status shouldBe SEE_OTHER
+        result.header("location") shouldBe Some(s"http://localhost:11111/income-through-software/return/$taxYear/view")
+      }
+
+      "Redirect to educational services question page when there is educational services amount but has no educationalServicesQuestion" in {
+        implicit lazy val result: WSResponse = {
+          authoriseAgentOrIndividual(isAgent = false)
+          dropEmploymentDB()
+          val medicalChildcareEducationModel = fullMedicalChildcareEducationModel.copy(educationalServicesQuestion = None, educationalServices = Some(11.0))
```

Comment:
you dont need to copy the new educationalServices value, the fullMedicalChildcareEducationModel already has one

Reply:
I put it deliberately as it makes it more obvious what we are testing and what the conditions are.